recipes/utf8.h/all/test_package/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+find_package(utf8.h REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} utf8.h::utf8.h)
```

Comment:
As you wish, but I would not recommend to show non official CMake config files or imported targets in test_package.

Reply:
I see, here we have a _problem_. We want to push existing packages and consumers to start using targets (right now, moving to `cmake_find_package[_multi]` generator), what to do with these libraries that don't have official or unofficial targets?

I would really like to know what you think cc/ @uilianries @SSE4 , maybe we are wrong trying to _force_ using targets and/or config files?

Reply:
I see. It's fine for me, I just try to use `cmake_find_package[_multi]` in test_package when we properly mimic an official find or config module file, but there is no "rule".

Reply:
You can run `conan install xxxx --generator markdown` and it will show some information, among that information it is how to consume the library using _common_ generators.

Reply:
I know, but it doesn't say if this is a proper emulation of official CMake config or module files. Looking at this markdown, consumer can't really know if he can rely on these informations for transparent integration or not. He still have to lost time to dig into original project to figure out if we properly emulate those informations or not.

Reply:
Understood, but the only alternative would be to read the actual source from `test_package/CMakeLists.txt` with its comments (assuming we are commenting if the integration is using official targets or not).. and that's not possible.

Reply:
Typically for these cases we just opt for the `${CONAN_LIBS}`

there's nothing in the recipe adding something special to the cmake find generators ... why test the Conan implementation here? I think it's easier to test the recipe rather than how it might be consumed.

Reply:
the projects which don't originate from the CMake (e.g. use another build system) usually don't have official module/target names. still, CMake documentation states that these projects could be imported to the CMake:
https://cmake.org/cmake/help/v3.20/manual/cmake-developer.7.html#find-modules
https://cmake.org/cmake/help/v3.20/manual/cmake-packages.7.html#creating-packages
so, if foreign projects follow the sane CMake naming conventions, they could still be used in CMake, even without an official target.
as soon as official target added later, conan package could be updated to provide an alias with build modules.

Reply:
If some generators like `cmake` and `cmake_find_package[_multi]` get deprecated in the future in favor of `CMakeDeps` (https://github.com/conan-io/conan/pull/8568) and the CMakeToolchain, we need recipes to be prepared for that migration: the targets will be available in `CMakeDeps`, but `${CONAN_LIBS}` won't...